### PR TITLE
account: dump extra information on account or newly created service principal

### LINKED ISSUE #1097: Build out alternative authentication for SDKs
The Java and C# SDKs use the auth format as described here:
https://github.com/Azure/azure-sdk-for-java/blob/master/AUTH.md

We should generate this file when users login to enable single-sign-on across SDKs.

### PR DESCRIPTION
**Pull request by yugangw-msft**
Per ask from java sdk team. Fix #1097. I updated some naming so this whole thing won't look odd.
Once we are fine with the change, i will add one test

### CODE REVIEW

**Comment by yugangw-msft** [comment]
```bash
az account show --expanded-view
{
  "endpoints": {
    "activeDirectory": "https://login.microsoftonline.com",
    "activeDirectoryGraphResourceId": "https://graph.windows.net/",
    "activeDirectoryResourceId": "https://management.core.windows.net/",
    "gallery": "https://gallery.azure.com/",
    "management": "https://management.core.windows.net/",
    "resourceManager": "https://management.azure.com/",
    "sqlManagement": "https://management.core.windows.net:8443/"
  },
  "environmentName": "AzureCloud",
  "id": "a7d7a534-7494-4099-8350-b44e96c5xxxx",
  "name": "https://yugangwlogin",
  "password": "VerySecret-xxxxx",
  "tenantId": "26833e2e-ee80-4a9d-b24c-ffe135b7xxxx"
}
```

**Comment by yugangw-msft** [comment]
//cc:@derekbekoe @jianghaolu

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 import dateutil.parser
 
 from azure.cli.core._util import CLIError, todict, get_file_json
+from azure.cli.core._profile import Profile
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py]: For perf. reasons, can we move this import to inside the method it's used in.

importing Profile was v. slow in my experience.

Comment (derekbekoe) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py]: FYI on importing Profile:
```
$ time python -c 'from azure.cli.core._profile import Profile'

real	0m0.446s
user	0m0.364s
sys	0m0.080s
```

**Pull request merged**